### PR TITLE
feat(curriculum): Create TypeScript Quiz

### LINKED ISSUE #56227: [HACKTOBERFEST 2024] - TypeScript Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the TypeScript quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-typescript`.

**All quiz questions need to relate to one of these topics:**

- What is Typescript
- Why use it
- Different types in TS (union types, interfaces, etc)
- Generic types
- type narrowing
- tsconfig file

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by Nycto-c05**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56227

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 1]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 ### --question--
 
 #### --text--
 
-Placeholder question
+    function sum(a: number, b: number): number {
+	    return a + b;
+	}
+	console.log(sum(2, 3));
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:89]: code blocks need to be fenced, and here you also need words for your question, a code snippet is not enough to understand what it is being asked
````suggestion
Question using words goes here

```ts
    function sum(a: number, b: number): number {
	    return a + b;
	}
	console.log(sum(2, 3));

```
````

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 2]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder question
+What will be the output of the following TypeScript code?
+
+    enum Color {
+    	 Red,
+    	 Blue
+    }
+    console.log(Color.Red, Color.Blue);
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:430]:
```suggestion
What will be the output of the following TypeScript code?

```ts
    enum Color {
    	 Red,
    	 Blue
    }
    console.log(Color.Red, Color.Blue);
```

```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 3]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder distractor 3
+By using the `any` type
 
 #### --answer--
 
-Placeholder answer
+By adding a `?` after the parameter name
+
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:470]:
```suggestion
By adding a `?` after the parameter name

```

Comment (Nycto-c05) [reply curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md]: Got it! I've made the suggested changes. Thanks

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 4]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To define the structure of an object.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT a basic type in TypeScript?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:137]:
```suggestion
Which of the following is NOT a primitive type in TypeScript?
```
Just a suggestion, but the word primitive seems more accurate.

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 5]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder question
+What will be the output of the following TypeScript code?
 
+```ts
+    enum Color {
+    	 Red,
+    	 Blue
+    }
+    console.log(Color.Red, Color.Blue);
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:433]:
````suggestion
```ts
enum Color {
  Red,
  Blue,
}
console.log(Color.Red, Color.Blue);
```
````

The code block isn't closed, and the formatting is off.

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md (hunk 1 above)
Comment (lasjorg) [reply curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:92]: The code needs to be formatted as well.

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md (hunk 1 above)
Comment (Nycto-c05) [reply curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md]: Fixed the formatting. Thanks for pointing it out. @lasjorg

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 6]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Why should TypeScript be used in a project?
 
 #### --distractors--
 
-Placeholder distractor 1
+Eliminates need for testing in applications.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:46]:
```suggestion
It eliminates the need for testing in applications.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 7]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It makes styling websites easier.
 
 #### --answer--
 
-Placeholder answer
+Improves maintainability by adding type definitions to JavaScript.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:58]:
```suggestion
It improves maintainability by adding type definitions to JavaScript.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 8]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+C++
 
 #### --answer--
-
-Placeholder answer
+JavaScript
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:79]:
```suggestion

JavaScript
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 9]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
+function sum(a: number, b: number): number {
+  return a + b;
+}
+console.log(sum(2, 3));
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+ `2 + 3`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:96]:
```suggestion
`2 + 3`
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 10]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+To create a new data type.
 
 ---
 
-Placeholder distractor 3
+To create a new JavaScript Object
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:126]:
```suggestion
To create a new JavaScript Object.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 11]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 2
+To define structure of an object.
 
 ---
 
-Placeholder distractor 3
+To create a new data type
+
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:259]:
```suggestion
To create a new data type.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 12]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which of the following is a **proper example** of a generic function in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+	function add(a: number, b: number): number { return a + b; }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:273]:
```suggestion
`function add(a: number, b: number): number { return a + b; }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 13]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+	function multiply(a: number, b: number): { return a * b; }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:277]:
```suggestion
`function multiply(a: number, b: number): { return a * b; }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 14]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+	function getValue<T>(value: any): any { return value; }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:281]:
```suggestion
`function getValue<T>(value: any): any { return value; }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 15]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+	function identity<T>(value: T): T { return value; }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:285]:
```suggestion
`function identity<T>(value: T): T { return value; }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 16]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --text--
 
-Placeholder question
+What is type narrowing in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+
+Reducing the size of compiled JavaScript
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:296]:
```suggestion
Reducing the size of compiled JavaScript
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 17]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Limiting the number of types in a union
 
 ---
 
-Placeholder distractor 3
+Removing unused types from a TypeScript project.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:304]:
```suggestion
Removing unused types from a TypeScript project
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 18]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+To write Code that narrows down a union type to a more specific type
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:308]:
```suggestion
To write code that narrows down a union type to a more specific type
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 19]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+`package.json`
 
 ---
 
-Placeholder distractor 2
+`package.json`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:322]: These should not be the same.

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 20]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+The specific module system to use for imports and exports.
 
 #### --answer--
 
-Placeholder answer
+
+The ECMAScript version to which the TypeScript compiler will transpile the code.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:353]:
```suggestion
The ECMAScript version to which the TypeScript compiler will transpile the code.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 21]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
-Placeholder question
+What will be the output of the following TypeScript code?
 
+```ts
+enum Color {
+  Red,
+  Blue,
+}
+console.log(Color.Red, Color.Blue);
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:433]:
````suggestion
```

````

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 22]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+By using the `any` type
 
 #### --answer--
 
-Placeholder answer
-
+By adding a `?` after the parameter name
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:470]:
```suggestion
By adding a `?` after the parameter name

```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
There's a few fixes here, but you're getting closer!

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md (hunk 19 above)
Comment (Nycto-c05) [reply curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md]: Fixed it. Thanks for pointing it out.

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 23]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Creating intersection types
 
 #### --answer--
 
-Placeholder answer
+Implementing a class
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:241]: Not exactly sure what is meant by, implementing a class?
If it is referring to the use of `extends` could we be more explicit about it?

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 24]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It prevents using the `any` type in TypeScript code.
 
 #### --answer--
 
-Placeholder answer
+It warns if a function does not return a value.
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:373]:
```suggestion
Checks all code paths in a function to ensure they return a value.
```
Might be a worthwhile distinction to add.

All functions implicitly return `undefined` and you can implicitly/explicitly have `undefined` as the return type, and Typescript will let the function "bottom out" without an explicit return.

---

Works
```
function isTrue(test: boolean) {
  if (test) {
    console.log("true");
  } else {
    return undefined;
  }
  console.log("Bottom");
}
isTrue(true);
```
Doesn't work
```
function isTrue(test: boolean) {
  if (test) {
    console.log("true");
  } else {
    return false;
  }
  console.log("Bottom");
}
isTrue(true);
```

**Comment by naomi-lgbt** [comment]
@Nycto-c05 I have already approved your quiz. It is now waiting for an approval from a second maintainer.

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md (hunk 23 above)
Comment (Nycto-c05) [reply curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:241]: Thanks for pointing it out. I should've phrased that better.
What I meant was that classes can implement interfaces or Interfaces can be used to declare class implementations.

Thanks again

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md (hunk 24 above)
Comment (Nycto-c05) [reply curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:373]: Got it. Learnt something new today.
Added the suggested changes

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 25]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What is the primary purpose of the `type` keyword in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+To declare variables
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:185]:
```suggestion
To declare variables.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 26]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+To create classes
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:189]:
```suggestion
To create classes.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 27]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+To implement interfaces
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:193]:
```suggestion
To implement interfaces.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 28]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+To define custom types
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:197]:
```suggestion
To define custom types.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 29]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which of the following capabilities is unique to `interface` and CANNOT be achieved using `type` in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+Defining object shapes
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:229]:
```suggestion
Defining object shapes.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 30]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Creating union types
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:233]:
```suggestion
Creating union types.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 31]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Creating intersection types
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:237]:
```suggestion
Creating intersection types.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 32]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+Declaring class implementations
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:241]:
```suggestion
Declaring class implementations.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 33]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What is type narrowing in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+Reducing the size of compiled JavaScript
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:295]:
```suggestion
Reducing the size of compiled JavaScript.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 34]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Limiting the number of types in a union
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:299]:
```suggestion
Limiting the number of types in a union.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 35]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Removing unused types from a TypeScript project
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:303]:
```suggestion
Removing unused types from a TypeScript project.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 36]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+To write code that narrows down a union type to a more specific type
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:307]:
```suggestion
To write code that narrows down a union type to a more specific type.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 37]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+How can you specify that a function parameter is optional in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+By using the `optional` keyword
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:457]:
```suggestion
By using the `optional` keyword.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 38]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+By setting a default value
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:461]:
```suggestion
By setting a default value.
```

File: curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md [hunk 39]
```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+By using the `any` type
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md:465]:
```suggestion
By using the `any` type.
```

**Review by lasjorg — CHANGES_REQUESTED** [review]
Just some punctuation inconsistencies.

**Review by lasjorg — CHANGES_REQUESTED** [review]

**Review by lasjorg — APPROVED** [review]

**Pull request merged**